src/DistributedTracingManager.ts

```diff
@@ -0,0 +1,251 @@
+        let registry = iothub.Registry.fromConnectionString(iotHubConnectionString);
+
+        let mode: boolean = undefined;
+        let samplingRate: number = undefined;
+        let twin;
+
+        if (deviceIds.length === 1) {
+            twin = await Utility.getTwin(registry, deviceIds[0]);
+
+            if (twin.properties.desired[Constants.DISTRIBUTED_TWIN_NAME]) {
+                mode = Utility.parseDesiredSamplingMode(twin);
+                samplingRate = Utility.parseDesiredSamplingRate(twin);
+            }
+
+            if (updateType === DistributedSettingUpdateType.OnlySamplingRate) {
+                mode = undefined;
+            }
+
+            if (updateType === DistributedSettingUpdateType.OnlyMode) {
+                samplingRate = undefined;
+            }
+        }
+
+        if (updateType !== DistributedSettingUpdateType.OnlySamplingRate) {
+            let selectedItem: SamplingModeItem = await vscode.window.showQuickPick(
+                this.getSamplingModePickupItems(),
+                { placeHolder: "Select whether to enable/disable the distributed tracing...", ignoreFocusOut: true },
+            );
+            if (!selectedItem) {
+                return;
+            }
+            mode = selectedItem.distributedTracingEnabled;
+        }
+
+        if (updateType !== DistributedSettingUpdateType.OnlyMode) {
+            if (mode !== false) {
+                samplingRate = await this.promptForSamplingRate(`Enter sampling rate, integer within [0, 100]`, samplingRate);
+
+                if (samplingRate === undefined) {
+                    return;
+                }
+            }
+        }
+
+        TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingStartEvent);
+
+        await vscode.window.withProgress({
+            title: `Update Distributed Tracing Setting`,
+            location: vscode.ProgressLocation.Notification,
+        }, async () => {
+            try {
+                const result = await this.scheduleTwinUpdate(mode, samplingRate, iotHubConnectionString, deviceIds);
+                TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingDoneEvent, { Result: "Success" }, iotHubConnectionString);
+                this.outputLine(Constants.IoTHubDistributedTracingSettingLabel,
+                    `Update distributed tracing setting for device [${deviceIds.join(",")}] complete!` + result);
+            } catch (err) {
+                TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingDoneEvent, { Result: "Fail", Message: err.message }, iotHubConnectionString);
+                this.outputLine(Constants.IoTHubDistributedTracingSettingLabel, `Failed to get or update distributed setting: ${err.message}`);
+                return;
+            }
+        });
+    }
+
+    private async scheduleTwinUpdate(enable: boolean, samplingRate: number, iotHubConnectionString: string, deviceIds: string[]) {
```

Comment:
This function to better naming `updateDeviceTwin`

Reply:
fixed